src/OVAL/probes/independent/system_info.c

```diff
@@ -353,6 +353,21 @@ static ssize_t __sysinfo_saneval(const char *s)
 	return (ssize_t)real_length;
 }
 
+static FILE *_fopen_with_prefix(const char *prefix, const char *path)
+{
+	FILE *fp;
+	if (prefix != NULL) {
+		char path_with_prefix[PATH_MAX]; // PATH_MAX includes terminating '\0' byte
+		size_t prefix_len = strlen(prefix);
+		strncpy(path_with_prefix, prefix, PATH_MAX);
+		strncpy(path_with_prefix + prefix_len, path, PATH_MAX - prefix_len);
```

Comment:
cound't this be done in an easier way with `oscap_asprintf`? or are you doing it like this because you are worried of it running over PATH_MAX?

Reply:
It could.